### ISSUE TITLE
libsass compatibility?

### ISSUE THREAD

**Issue by ronilaukkarinen** [penman/Sass-Web-Fonts#14]
Hi,

I recently moved to the latest libsass and noticed this:

      [11:47:41] { [Error: /Users/rolle/test/web-fonts:91: @import directives are not allowed inside mixins and functions
      ]
        message: '/Users/rolle/test/web-fonts:91: @import directives are not allowed inside mixins and functions\n',
        showStack: false,
        showProperties: true,
        plugin: 'gulp-sass',
        __safety: { toString: [Function] } }

Any way to make this to work with libsass?

**Comment by Minopolis** [comment]
Apparently, we can pass variables to `@import` statements in libsass by using a function if we hack around a bit:

	//change this
	@mixin web-fonts($fonts...) {
	  $protocol: wf-protocol();
	  $query-string: wf-params-string($fonts);
	  $url: "#{$protocol}//fonts.googleapis.com/css?#{$query-string}";
	  @import url($url);
	}
	//to this
	@function web-fonts($fonts...) {
	  $protocol: wf-protocol();
	  $query-string: wf-params-string($fonts);
	  $url: "#{$protocol}//fonts.googleapis.com/css?#{$query-string}";
	  @return $url;
	}

	//and later on, you can use the function like this:

	@import url(web-fonts('Open Sans'));
	//generates://fonts.googleapis.com/css?family=Open%20Sans

	$web-fonts-params: (subset: "latin,latin-ext");
	@import url(web-fonts('Open Sans'));
	//generates://fonts.googleapis.com/css?family=Open%20Sans&subset=latin%2Clatin-ext

	@import url(web-fonts('Open Sans:400,700'));
	//generates: //fonts.googleapis.com/css?family=Open%20Sans%3A400%2C700

	@import url(web-fonts('Open Sans:400,700|Roboto:400,700'));
	//generates: //fonts.googleapis.com/css?family=Open%20Sans%3A400%2C700|Roboto%3A400%2C700

For some reason libsass doesn't notice us using variables this way.
Tested in Brackets with [Brackets SASS](https://github.com/jasonsanjose/brackets-sass) extension

**Comment by penman** [comment]
Oh, cool! I'll investigate later this week. :)

**Comment by ronilaukkarinen** [comment]
Nice findings @Minopolis! Thanks for keeping the possibility open.

**Comment by Minopolis** [comment]
@ronilaukkarinen Pleasure :)

**Comment by penman** [comment]
Okay, I've started working on this on the [libsass](https://github.com/penman/Sass-Web-Fonts/tree/libsass) branch. It's probably going to be possible to support libsass, with a slightly different syntax.

In future, I'll probably change the documentation so it recommends everybody uses the old syntax, but keep the new one around to preserve backwards compatibility.

**Comment by jgacuca567** [comment]
@penman, what was the solution for the answer, when I keep getting the error 

```Compass
@import directives are not allowed inside mixins and functions
```

**Comment by penman** [comment]
Sorry I've taken such a long time on this. I'm working on a new version of Sass-Web-Fonts that should be compatible with libsass. Is anybody available to help me test it?

**Comment by jgacuca567** [comment]
@penman what would be required to test?

**Comment by penman** [comment]
@jgacuca567 Just need you to try using a new version in one of your projects (libsass or Ruby Sass) and let me know if everything works

**Comment by ronilaukkarinen** [comment]
@penman Let me know when it's ready, I'd love to test it out.

**Comment by penman** [comment]
For anyone that wants to test, please take a look at https://github.com/penman/Sass-Web-Fonts/blob/2.0.0. (You can install it from bower with `bower install sass-web-fonts#2.0.0`) It should have enough information for you to start using it in your project. Let me know how it goes so I can get the new version released! :)

**Comment by penman** [comment]
Bah, got the URL wrong and can't edit my comment because I'm on my phone. Try https://github.com/penman/Sass-Web-Fonts/tree/2.0.0?files=1

**Comment by ronilaukkarinen** [comment]
@penman Works fine here!

````
$ node-sass --version
node-sass	3.3.2	(Wrapper)	[JavaScript]
libsass  	3.2.5	(Sass Compiler)	[C/C++]
````

**Comment by penman** [comment]
@ronilaukkarinen Excellent, thanks!

**Comment by jgacuca567** [comment]
@penman for Grunt Libsass, it is no longer being maintained. Currently Using grunt-contrib-sass

**Comment by penman** [comment]
@jgacuca567 Does it work with grunt-contrib-sass?

**Comment by jgacuca567** [comment]
Will have to get back on you on that, currently getting grunt errors right
now.

On Fri, Oct 30, 2015 at 12:32 PM Ross Penman <notifications@github.com>
wrote:

> @jgacuca567 <https://github.com/jgacuca567> Does it work with
> grunt-contrib-sass?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/penman/Sass-Web-Fonts/issues/14#issuecomment-152579540>
> .
>
-- 
James Gacuca
jgacuca@gmail.com
(202)-569-1159

**Comment by jgacuca567** [comment]
@penman No Errors So Far, after using grunt-contrib-sass.

**Comment by penman** [comment]
Excellent @jgacuca567! So that's working fine for one libsass user and one Ruby user. Should be good to release!

**Comment by penman** [comment]
I've created a [pre-release](https://github.com/penman/Sass-Web-Fonts/releases/tag/2.0.0-rc1) of this new version. I'd appreciate if anybody reading this thread could take the time to test it out and let me know how it goes (successful or unsuccessful). If nobody finds any bugs in the next 4 days (by 2015-11-03), I'll release it properly then.

**Comment by jgacuca567** [comment]
@penman The Upgrading Guide Redirects to a 404 Page.

**Comment by penman** [comment]
Oops. Fixed! Thanks @jgacuca567.

**Comment by ronilaukkarinen** [comment]
How about loading up multiple webfonts and multiple weights? It works fine for a single font, but I get `error reading values after )` and `invalid top-level expression` too easily. The syntax should be more readable, it's hard to focus with that many parentheses, don't you think?

**Comment by penman** [comment]
I agree that the syntax is not the best. Can you suggest an alternative that would preserve all the existing functionality? I can't think of one.

**Comment by ronilaukkarinen** [comment]
@penman Actually, I can't think of one either, just food for thought if anyone has ideas about it.

**Comment by penman** [comment]
@ronilaukkarinen I agree with you and welcome suggestions from anybody on how to make the syntax clearer. Now is the perfect time, too, since the libsass support requires breaking changes anyway.

**Comment by penman** [comment]
Right, I think it's time for a release!

**Comment by penman** [comment]
Okay, I have released Sass-Web-Fonts 2.0.0. Libsass is supported, and we can finally close this issue for good after a year. :D :D :D

**Comment by penman** [comment]
For those interested, [libsass can finally parse `@import url()` correctly](https://github.com/sass/libsass/pull/1809)
